Coordinate.cpp

```diff
@@ -1,11 +1,78 @@
 #include "Coordinate.h"
+#include <cmath>
 
 Coordinate::Coordinate() : x(0), y(0) {
 }
 
 Coordinate::Coordinate(int x, int y) : x(x), y(y) {
 }
 
+Coordinate::Coordinate(Direction d)
```

Comment:
I think I can replace Direction with DirectionVector here, it would make slightly more sense.

In a mathematical sense, all points are vectors to the origin, so behavior wise this system will hold.